client/driver/docker.go

```diff
@@ -1458,10 +1463,48 @@ func calculatePercent(newSample, oldSample, newTotal, oldTotal uint64, cores int
 	return (float64(numerator) / float64(denom)) * float64(cores) * 100.0
 }
 
+func authOptionFromDriverConfig(auth DockerDriverAuth) (*docker.AuthConfiguration, error) {
+	return &docker.AuthConfiguration{
+		Username:      auth.Username,
+		Password:      auth.Password,
+		Email:         auth.Email,
+		ServerAddress: auth.ServerAddress,
+	}, nil
+}
+
+func authOptionFromHelper(helperName, repoName string, clientConfig *config.Config) (*docker.AuthConfiguration, error) {
+	helper := dockerAuthHelperPrefix + helperName
+	cmd := exec.Command(helper, "get")
+	cmd.Stdin = strings.NewReader(repoName)
+
+	// For helper to be able to read credentials from vault
+	cmd.Env = append(os.Environ(), fmt.Sprintf("VAULT_TOKEN=%s", clientConfig.VaultConfig.Token))
```

Comment:
I considered this when writing the code. I get a feeling this deserve something bigger than a small helper-detail in the docker driver though. There must be more places where external processes/plugins are called for as a part of the task-setup, needing access to different nomad-config, for example the vault-address used, the consul-connection details etc?

From a security-standpoint however; the helper will run in the same security-context as the nomad process. This means whomever control the helper can still read the token from the nomad-config?

If it simplifies the approval of this PR, we could skip the custom env altogether, postpone it for something more thought-through?